### PR TITLE
edit/delete subjects

### PR DESCRIPTION
**Pull request by qmau94**

### CODE REVIEW

File: app/views/supervisor/subjects/_subject_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      <%= f.fields_for :tasks do |builder| %>
+        <%= render "task_fields", f: builder %>
+      <% end %>
+      <%= link_to_add_fields t("subjects.add_task"), f, :tasks %>
+
+      <%= f.submit t("subjects.#{params[:action]}_title"),
+        class: "btn btn-primary" %>
+    <% end %> 
+  </div>
+</div>
```
Comment (leduykhanhhedspi) [inline app/views/supervisor/subjects/_subject_form.html.erb]: end of file

File: app/controllers/supervisor/subjects_controller.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class Supervisor::SubjectsController < ApplicationController
   before_action :logged_in_user
   before_action :verify_supervisor
-  before_action :find_subject, only: [:show]
+  before_action :find_subject, except: [:index, :new, :create] 
   
   def new
     @subject = Subject.new
```
Comment (dhung133) [inline app/controllers/supervisor/subjects_controller.rb]: end

File: app/views/supervisor/subjects/index.html.erb [hunk 3]
```diff
@@ -16,6 +16,10 @@
             <%= render subject.tasks %>
           </ol>
         <% end %>
+        <%= link_to t("subjects.edit"), edit_supervisor_subject_path(subject),
+          class: "btn btn-success"%>
```
Comment (yongsokheng) [inline app/views/supervisor/subjects/index.html.erb]: thieu mot space

File: app/views/supervisor/subjects/index.html.erb [hunk 4]
```diff
@@ -16,6 +16,10 @@
...
+        <%= link_to t("subjects.delete"), supervisor_subject_path(subject),
+          method: :delete, class: "btn btn-danger"%>
```
Comment (yongsokheng) [inline app/views/supervisor/subjects/index.html.erb]: thieu mot space

File: app/views/supervisor/subjects/show.html.erb [hunk 5]
```diff
@@ -2,6 +2,10 @@
 <div class="col-xs-6 col-md-4 text-center">
   <h1><%= @subject.name %></h1>
   <h3><%= @subject.description %></h3>
+  <%= link_to t("subjects.edit"), edit_supervisor_subject_path(@subject),
+    class: "btn btn-success"%>
```
Comment (yongsokheng) [inline app/views/supervisor/subjects/show.html.erb]: thieu mot space

File: app/views/supervisor/subjects/show.html.erb [hunk 6]
```diff
@@ -2,6 +2,10 @@
...
+  <%= link_to t("subjects.delete"), supervisor_subject_path, method: :delete,
+    class: "btn btn-danger"%>
```
Comment (yongsokheng) [inline app/views/supervisor/subjects/show.html.erb]: thieu mot space

File: app/views/supervisor/subjects/show.html.erb [hunk 7]
```diff
@@ -2,6 +2,10 @@
...
 </div>
 <div class="col-xs-12 col-md-8">
 <h3><%= t "subjects.tasks" %> (<%= @subject.tasks.count %>)</h3>
```
Comment (yongsokheng) [inline app/views/supervisor/subjects/show.html.erb]: xoa ()

File: app/views/supervisor/subjects/edit.html.erb [hunk 8]
```diff
@@ -0,0 +1,3 @@
+<% provide :title, t("subjects.edit_title") %>
```
Comment (yongsokheng) [inline app/views/supervisor/subjects/edit.html.erb]: <%= %>

File: app/views/supervisor/subjects/new.html.erb [hunk 9]
```diff
@@ -1,23 +1,3 @@
...
-      <%= f.fields_for :tasks do |builder| %>
-        <%= render "task_fields", f: builder %>
-      <% end %>
-      <%= link_to_add_fields t("subjects.add_task"), f, :tasks %>
-
-      <%= f.submit t("subjects.create"), class: "btn btn-primary" %>
-    <% end %> 
-  </div>
-</div>
+<% provide :title, t("subjects.new_title") %>
```
Comment (yongsokheng) [inline app/views/supervisor/subjects/new.html.erb]: <%= %>

**Comment by yongsokheng** [comment]
OK

**Pull request merged**